### PR TITLE
Strip .git suffix from GitHub repository name as GitURL does

### PR DESCRIPTION
**Pull request by ikesyo**
Addresses https://github.com/Carthage/Carthage/issues/745#issuecomment-139491947.

The following is the current output before this PR with the input `github "https://github.com/danielgindi/ios-charts.git"` or `github "danielgindi/ios-charts.git"`:

```sh
*** Cloning ios-charts.git
A shell task failed with exit code 128:
fatal: could not read Username for 'https://github.com': terminal prompts disabled
```

### CODE REVIEW

File: Source/CarthageKit/GitHub.swift [hunk 1]
```diff
@@ -136,7 +136,7 @@ public struct GitHubRepository: Equatable {
 	public init(server: Server = .GitHub, owner: String, name: String) {
 		self.server = server
 		self.owner = owner
-		self.name = name
+		self.name = stripGitSuffix(name)
```
Comment (mdiep) [inline Source/CarthageKit/GitHub.swift]: I think it'd be better to call `stripGitSuffix` in the caller.

Comment (ikesyo) [inline Source/CarthageKit/GitHub.swift]: Thank you, I was unsure which is better.

**Comment by ikesyo** [comment]
Done! :zap:

**Comment by mdiep** [comment]
Looks great! :zap:

**Pull request merged**